### PR TITLE
Add additional information to policies

### PR DESCRIPTION
**Pull request by xurui-c**
https://getsentry.atlassian.net/browse/SNS-2796

The end goal is to have a payload that looks like the following:
```
"quota_allowance": {
	"summary": {
		"threads_used": 2, # the actual amount of max_threads sent to clickhouse
		"throttled_by": {
			"policy": "BytesScannedRejectingPolicy",
			"throttle_threshold": 800,
			"quota_used": 900,
			"quota_units": "bytes",
			"suggestion": "scan fewer bytes n00b",
		},
		"rejected_by": {},
	}
	# policy details as before
	...
}
```
In order to accomplish this, this PR adds in the relevant information into policy definitions and the `QuotaAllowance` class, and will be followed up by a subsequent PR which will actually update the payload to the above.

### CODE REVIEW

File: snuba/query/allocation_policies/bytes_scanned_rejecting_policy.py [hunk 1]
```diff
@@ -314,3 +311,71 @@ def _update_quota_balance(
...
+            customer_tenant_value,
+        ) = self._get_customer_tenant_key_and_value(tenant_ids)
+        referrer = tenant_ids.get("referrer", "no_referrer")
+        if referrer in _PASS_THROUGH_REFERRERS:
+            return sys.maxsize
+        return self.__get_scan_limit(
+            customer_tenant_key, customer_tenant_value, referrer
+        )
+
+    def get_quota_used(self, tenant_ids: dict[str, str | int]) -> int:
```
Comment (xurui-c) [inline snuba/query/allocation_policies/bytes_scanned_rejecting_policy.py:337]: @volokluev I'm not a fan of repeating this chunk of code

File: snuba/query/allocation_policies/__init__.py [hunk 2]
```diff
@@ -799,6 +804,26 @@ def _update_quota_balance(
...
+        pass
+
+    @abstractmethod
+    def get_quota_units(self) -> str:
+        pass
+
+    @abstractmethod
+    def get_suggestion(self) -> str:
+        pass
+
```
Comment (volokluev) [inline snuba/query/allocation_policies/__init__.py:826]: why is this necessary?

File: snuba/query/allocation_policies/bytes_scanned_rejecting_policy.py [hunk 3]
```diff
@@ -314,3 +311,71 @@ def _update_quota_balance(
...
+                    ],
+                ),
+            ]
+        )
+
+        used_quota = 0
+        for granted_quota in granted_quotas:
+            used_quota += granted_quota.granted
+
+        return used_quota
```
Comment (volokluev) [inline snuba/query/allocation_policies/bytes_scanned_rejecting_policy.py:375]: Is this now running the calculation twice?

**Review by volokluev — COMMENTED** [review]
It's not clear to me why you need to change anything besides the QuotaAllowance class

File: snuba/query/allocation_policies/bytes_scanned_rejecting_policy.py (hunk 1 above)
Comment (volokluev) [reply snuba/query/allocation_policies/bytes_scanned_rejecting_policy.py:337]: so why are you doing it then?

**Comment by xurui-c** [comment]
redesign
